application/src/main/java/org/opentripplanner/standalone/configure/ConstructApplication.java

```diff
@@ -334,4 +336,12 @@ public EmissionsDataModel emissionsDataModel() {
   public StreetLimitationParameters streetLimitationParameters() {
     return factory.streetLimitationParameters();
   }
+
+  private void initializeSchema() {
```

Comment:
Is the method name correct? To me it looks like this only validate the schema. The schema used else where is created by Dagger? I am not particular found of this - this defeats part of the point of DI. Instead I think there should be unit-tests to test schema creation. Not sure what can go wrong when you start the server and create the schema based on custom config - that is not testable. Also, the schema should be created eagerly (not lazy) by Dagger and the server should go down almost tha same way as you enforce here - or did I miss something?

Reply:
I'm pretty sure that I tested that this does initialize the schema and it is not recreated ever again. That's why I named it like this.

Since the schema creation depends on configuration now, I didn't want to end up in a situation where the server starts up but can't serve GraphQL traffic due to some NPE for example. Ofc, it's possible to avoid those types of issues in other ways as well (thorough unit testing, null checks and/or wrapping the default value creation in a try catch).